src/extension.ts

```diff
@@ -184,6 +192,50 @@ export async function activate(context: vscode.ExtensionContext): Promise<IPower
     };
 }
 
+/** Registers a command that waits for PSES Activation and returns the PID, used to auto-attach the PSES debugger */
+function registerWaitForPsesActivationCommand(context: vscode.ExtensionContext) {
+    return vscode.commands.registerCommand(
+        "PowerShell.WaitForPsesActivationAndReturnProcessId",
+        async () => {
+            const pidFileName = `PSES-${vscode.env.sessionId}.pid`;
+            const pidFile = vscode.Uri.joinPath(context.globalStorageUri, "sessions", pidFileName);
+            const fs = vscode.workspace.fs;
+            // Wait for the file to be created
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition, no-constant-condition
+            while (true) {
+                try {
+                    const pidContent = await fs.readFile(pidFile);
+                    const pid = parseInt(pidContent.toString(), 10);
+                    try {
+                        // Check if the process is still alive, delete the PID file if not and continue waiting.
```

Comment:
I'm not sure I follow this logic. I _think_ it's to kill an old copy of PSES, but when might that be necessary?

Reply:
https://nodejs.org/api/process.html#process_process_kill_pid_signal

It's an obtuse node trick, I'll add a comment clarifying.

Reply:
Oh WTF how weird. TIL.